employee-rostering-frontend/src/ui/header/Toolbar.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Button,
+  ButtonVariant,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  Toolbar,
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core';
+import { BellIcon, CogIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { tenantOperations } from 'store/tenant';
+import { AppState } from 'store/types';
+import Tenant from 'domain/Tenant';
+import TenantRestServiceClient from 'services/TenantRestServiceClient';
+
+interface StateProps {
+  currentTenantId: number;
+  tenantList: Tenant[];
+}
+
+const mapStateToProps = ({ tenantData }: AppState): StateProps => ({
+  currentTenantId: tenantData.currentTenantId,
+  tenantList: tenantData.tenantList
+});
+
+interface ToolbarState {
+  isTenantSelectOpen: boolean;
+}
+
+export interface DispatchProps {
+  refreshTenantList: typeof tenantOperations.refreshTenantList;
+  changeTenant: typeof tenantOperations.changeTenant;
+}
+
+const mapDispatchToProps: DispatchProps = {
+  refreshTenantList: tenantOperations.refreshTenantList,
+  changeTenant: tenantOperations.changeTenant
+};
+
+export type Props = StateProps & DispatchProps;
+
+class ToolbarComponent extends React.Component<Props, ToolbarState> {
+  tenantRestService: TenantRestServiceClient = new TenantRestServiceClient();
+  toolbarState: ToolbarState;
+
+  constructor(props: Props) {
+    super(props);
+    this.toolbarState = { isTenantSelectOpen: false };
+  }
+
+  componentDidMount() {
+    this.props.refreshTenantList();
+  }
+
+  setCurrentTenant(newTenantId: number) {
+    this.toolbarState = {
+      isTenantSelectOpen: false
+    };
+    this.props.changeTenant(newTenantId);
+  }
+
+  setIsTenantSelectOpen(isOpen: boolean) {
+    this.toolbarState = { isTenantSelectOpen: isOpen };
+    this.refresh();
+  }
+
+  refresh() {
+    this.setState(this.toolbarState);
+  }
+
+  render() {
+    if (this.props.tenantList.length === 0) {
+      return <Toolbar><ToolbarGroup /> <ToolbarGroup>
```

Comment:
Minor problem but the second ToolbarGroup with bell and cog is duplicated in both branches.